frontend/public/components/filter-toolbar.tsx

```diff
@@ -167,19 +167,23 @@ const FilterToolbar_: React.FC<FilterToolbarProps & RouteComponentProps> = (prop
   // Map row filters to select groups
   const dropdownItems = generatedRowFilters.map((rowFilter) => (
     <SelectGroup key={rowFilter.filterGroupName} label={rowFilter.filterGroupName}>
-      {rowFilter.items?.map?.((item) => (
-        <SelectOption
-          data-test-row-filter={item.id}
-          key={item.id}
-          inputId={item.id}
-          value={item.id}
-        >
-          <span className="co-filter-dropdown-item__name">{item.title}</span>
-          <Badge key={item.id} isRead>
-            {item.count}
-          </Badge>
-        </SelectOption>
-      ))}
+      {rowFilter.items?.map?.((item) =>
+        item.id === 'system' && item.count === 0 ? (
+          <div></div>
+        ) : (
```

Comment:
I agree this is less than ideal and I struggled with this for a while.  There is a requirement for the requester filter that if there are no system namespaces, that option isn't shown at all.  

Ideally  on frontend > public > components > namespace.jsx starting on line 104: `export const REQUEST_FILTER = { me: 'me', user: 'user', system: 'system' };` the system option shouldn't be included if there are no system namespaces/projects.

The problem is that we don't know how many system namespaces/project there are until much deeper in the stack inside of filter-toobar.tsx.  I suppose we can get the items and try to do a count within namespace.jsx just to see if any exist but namespace.jsx is kind of beast to begin with and it may mean duplicating code.

Reply:
> There is a requirement for the requester filter that if there are no system namespaces, that option isn't shown at all.

OK, I see. That's inconsistent with other filters, so we might want to look at that more generally. I'm inclined to say we should be consistent everywhere and not hide it.

If we really want that behavior, we should be able to add a property into the filter definition like

```ts
      { id: REQUEST_FILTER.system, title: 'System', hideIfNoMatches: true },
```

and have the filter toolbar look for that prop.

Reply:
Ohhh.  I like the idea of passing a property.  I'll look into adding it.

Reply:
There are some edge cases where the filter option could appear/disappear while you have the dropdown open if your access to a system project changes when on the page. Or you could have previously selected `System` as a filter and the option gets removed when your access to the system project is removed.

I'd say we should hold off on that behavior for now and consider it more generally.

Reply:
@gdoyle1 Thoughts on not hiding the "system" option in the Requestor filter if no system projects/namespaces are in the table?   So in the screenshot below, `System (0)` would be shown.
<img width="184" alt="RequesterFilter" src="https://user-images.githubusercontent.com/82059948/122255916-35eae400-ce94-11eb-9c13-6a8f47286d45.png">

See Sam's thoughts above.

Reply:
@kdoberst If there are no system projects in the table, then I think it's fine to hide the "system" option completely, yes! The only filter options that should populate there are ones that are living in the table @kdoberst